core/src/main/java/io/undertow/server/handlers/accesslog/DefaultAccessLogReceiver.java

```diff
@@ -153,75 +154,66 @@ public void logMessage(final String message) {
      */
     @Override
     public void run() {
+        //check if we can transition to 2. If so, perform tasks in small chunks and check this.closed.
+        //move into 3 if(this.closed) and terminate run()
         if (!stateUpdater.compareAndSet(this, 1, 2)) {
             return;
         }
-        if (forceLogRotation) {
-            doRotate();
```

Comment:
I've been thinking about this, and if you remove this, you're breaking the rotate() method contract

Reply:
My bad, there is a check right below, so the rotation algorithm is working correctly in the PR.